service/frontend/workflowHandler.go

```diff
@@ -1941,11 +1941,11 @@ func (wh *WorkflowHandler) GetWorkflowExecutionHistory(
 		if err != nil {
 			return nil, wh.error(errInvalidNextPageToken, scope, getWfIDRunIDTags(wfExecution)...)
 		}
-		if execution.RunID != nil && execution.GetRunID() != token.RunID {
+		if execution.RunID != "" && execution.GetRunID() != token.RunID {
```

Comment:
hmm.  these handlers are... rather painful to read.  there's a lot of:
- random flipping between nil and zero-value-getters for the same fields
- checks of .GetSomething() values before checking if the request itself is nil, which is also an error
- checks like this, which probably belong a lot earlier in the method to ensure the request is sane before doing things.  e.g. this appears to be relevant _immediately_ after the `if !getRequest.GetSkipArchival() {`, but it's nearly 60 lines later.

and there are _four thousand lines_ like this in this file alone.

wat.